### PR TITLE
Integrate a Code of Conduct

### PR DESCRIPTION
**Pull request by lomky**
I rule and have completed some work on Case Manager that's ready for review!

As we discussed in Slack, we don't have a Code of Conduct in our project and think it's a darn nice idea to have one!

This pull request makes the following changes:
 - created the template of the Contributor Covenant
 - inserted Colin's email as PoC

To be done before merge:

  - chat about any extensions needed for our specific cause
  - establish how we should respond to any reports, as the [source advises](http://contributor-covenant.org/#enforcement)

No issue.

@colinxfleming I'm not sure who all to tag in for a conversation, but I trust you do :smile:

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+* Trolling, insulting/derogatory comments, and personal or political attacks
+* Public or private harassment
+* Publishing others' private information, such as a physical or electronic
+  address, without explicit permission
+* Other conduct which could reasonably be considered inappropriate in a
+  professional setting
+
+## Our Responsibilities
+
+Project maintainers are responsible for clarifying the standards of acceptable
```
Comment (colinxfleming) [inline CODE_OF_CONDUCT.md]: I think it's fine to name me and @mebates specifically here. Maybe `Project maintainers (@colinxfleming and @mebates) are responsible for ... `

File: CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+when an individual is representing the project or its community. Examples of
+representing a project or community include using an official project e-mail
+address, posting via an official social media account, or acting as an appointed
+representative at an online or offline event. Representation of a project may be
+further defined and clarified by project maintainers.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting the project team at c3flemin@gmail.com. The project team
```
Comment (colinxfleming) [inline CODE_OF_CONDUCT.md]: You can probably name me here -- Colin Fleming at c3flemin@gmail.com

**Comment by colinxfleming** [comment]
Going to tag in a bunch of us here -- @mebates @lwaldsc @NerdyGirl537 @vtmeghan @Kevin-Wei @DarthHater !

**Comment by colinxfleming** [comment]
To the question of enforcement: 

* I think enforcement procedure as its laid out here (a rep investigates, discusses with the team, and then takes an action, probably a warning or ban if circumstances call for it) is reasonable.

* I think @lomky 's insight about having an 'abortion issue clause' is a great, great idea. Maybe something to the effect of:

> This project deals explicitly with abortion access; in addition to treating each other respectfully, we commit to treat the individuals we serve by the same standards. ( ... etc ) 

Thoughts?

**Comment by lwaldsc** [comment]
This is a fantastic idea, especially with the project getting attention at
the hackathon and railscon. Seconding the thank you thank you thank you!

Starting a new job this week but will try and chime in when I get a second!

**Comment by Kevin-Wei** [comment]
👍  looks good to me

**Comment by NerdyGirl537** [comment]
Looks good to me.

**Comment by lwaldsc** [comment]
Scanned through it quickly and it's a perfect start. Agreed with Colin that
we need something that explicitly states that we are working with abortion.
Language he suggested works for me.

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (lomky) [inline CODE_OF_CONDUCT.md]: Seems like good editorial practice. 

Specify on the first instance, and leave as just "Project maintainers" throughout the rest?

File: CODE_OF_CONDUCT.md (hunk 2 above)
Comment (lomky) [inline CODE_OF_CONDUCT.md]: done!

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (colinxfleming) [inline CODE_OF_CONDUCT.md]: Sounds like a good call to me!

**Comment by lomky** [comment]
@colinxfleming I like that addition. I think I would explicitly add the Case Managers there as well. I think we all consider them a part of the team as a given, but someone contributing from afar may not be as clear on that / understand their role fully.

Modded suggestion: 

> This project deals explicitly with abortion access; in addition to treating our fellow developers respectfully, we commit to treat the case managers and the individuals we serve by these same standards.

And regardless of that tweak, where do we put this?  
Does this go under "Our Standards", "Scope", or is it important enough to have its own top level? My gut feeling is a subsection under "Our Standards" after the lists of behaviors.

**Comment by colinxfleming** [comment]
I think it makes the most sense to put that at the bottom of 'Our Standards', since it explicitly refers to them.

Love that full thing with your edits @lomky  - short and to the point and very clear! Have a slight suggested tweak on the case managers and individuals front -- (mostly just changing `individuals` to patients`) -- your thoughts? 

> This project deals explicitly with abortion access; in addition to treating our fellow developers respectfully, we commit to treat case managers and the patients they serve by these same standards.

**Comment by DarthHater** [comment]
@lomky @colinxfleming I would change "developers" to something more abstract like "contributors" in that example.

Love this. Nothing is coming to mind in terms of additions.

**Comment by lomky** [comment]
@colinxfleming @DarthHater I like both of those changes.

I remembered what I was initially thinking about for our abortion related tweaks.

The Code of Conduct covers trolling and political attacks, but we are left open to politely-worded-but-distracting discussions arguing things such as discussion of pro-life vs pro-choice, when life begins, etc.

So what I was thinking is adding a few examples under "Examples of unacceptable behavior..." basically establishing that this project is not an appropriate venue for discussion of the political, religious, or moral implication of abortion. 

tl;dr: I'm wary of the "false polite" trolls.

So I'm suggesting brainstorming for other examples of inappropriate behavior or discussions that have been used in the past to harass, but we don't have a good finger on in the Code of Conduct as written. Then so long as those topics are outside the need to discuss for project development, we can list them explicitly as not acceptable here.

**Comment by DarthHater** [comment]
No idea how to tackle that but I agree that it's a potential problem. Hard to codify though.

**Comment by colinxfleming** [comment]
@lomky I think the part of the standards involving trolling kinda covers it, but we can def add an additional clause for clarity after the bullets. How about this? 

> This is an explicitly pro-choice project serving pro-choice organizations; it is not an appropriate venue for broader discussions about the moral, political, or religious implications of abortion or abortion access.

**Comment by lomky** [comment]
That is awesome @colinxfleming. Your wordsmithing is :100: 

Given we have the Hackathon coming up this weekend, let's aim to get our first cut merged in by Friday afternoon.

It should be a living document, so we'll have the chance to do more as we continue.

**Comment by lwaldsc** [comment]
"This is an explicitly pro-choice project serving pro-choice organizations"

$$$$$$$$$$$$$$$$$$

**Comment by lomky** [comment]
Branch up to date with suggestions as of this comment :)

Please keep 'em comin'! This is great. :tada:

**Comment by stvnrlly** [comment]
I think the comment about settled questions is great and possibly something that more projects should consider. It may be useful to think of it as a policy stack like a tech stack: decisions are made with care early on and major changes to the stack later on are distractions at best.

Separately, I don't know how self-contained you would like this code to be, but it might be useful to accept reports through email to the Code for DC captains, contact form, or a more specific reporting system as an additional reporting/escalation option.

**Comment by lomky** [comment]
I just realized that I hadn't linked the base code of conduct are using:

https://github.com/ContributorCovenant/contributor_covenant

**Comment by colinxfleming** [comment]
@kscoulter flagged that explicit escalation steps would be helpful. 

Here's some draft language to work off. @kscoulter would love your thoughts!

```
## What to do if there's an incident

If you would like to report or flag an incident, please reach out to one or both of the project maintainers @colinxfleming and @mebates. You can do this anonymously if you prefer.
```

**Comment by colinxfleming** [comment]
Just a flag that I'm gonna smash that merge button @ 9pm EST! Excited to get this in!

**Comment by mebates-USDS** [comment]
Hey all! Sorry to pipe in late. I love this, great work! Very happy with the two sentences specific to our project topic. I agree with @stvnrlly that perhaps we should provide a venue for escalation that _isn't_ one of the project leads.

**Comment by mebates** [comment]
Hey all! Sorry to pipe in late. I love this, great work! Very happy with the two sentences specific to our project topic. I agree with @stvnrlly that perhaps we should provide a venue for escalation that _isn't_ one of the project leads.
